### PR TITLE
fix: infinite TTLs are not supported anymore, forward ttl=0

### PR DESCRIPTION
**Pull request by moritzraho**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by MichaelGoberling — APPROVED** [review]

File: lib/AdobeState.js [hunk 1]
```diff
@@ -317,20 +320,31 @@ class AdobeState {
...
+    const { ttl } = options
+    const { valid, errors } = validate(schema, { key, value, ttl })
     if (!valid) {
       logAndThrow(new codes.ERROR_BAD_ARGUMENT({
         messageValues: utils.formatAjvErrors(errors),
-        sdkDetails: { key, value, options, errors }
+        sdkDetails: { key, valueLength: value.length, options, errors }
+      }))
+    }
+    if (ttl !== undefined && (ttl < 0 || ttl > MAX_TTL_SECONDS)) {
```
Comment (shazron) [inline lib/AdobeState.js:339]: too bad AJV validator can't have a min max :/

**Review by shazron — APPROVED** [review]

File: lib/AdobeState.js (hunk 1 above)
Comment (moritzraho) [reply lib/AdobeState.js:339]: yes AJV has a min max, but the error message is not nice... 
```
AioCoreSDKError [AdobeStateLibError]: [AdobeStateLib:ERROR_BAD_ARGUMENT] WARNING: keyword 'minimum' was not handled for formatting: {
  "instancePath": "/ttl",
  "schemaPath": "#/properties/ttl/minimum",
  "keyword": "minimum",
  "params": {
    "comparison": ">=",
    "limit": 0
  },
  "message": "must be >= 0"
}
```

I wanted to make it clear that infinite ttls are not supported anymore!

Comment (shazron) [reply lib/AdobeState.js:341]: actually we can add the support in our code here, but maybe for next time: https://github.com/adobe/aio-lib-state/blob/94e34ad9f4286de105ecda8fd8d917d1f1102b99/lib/utils.js#L63

Comment (moritzraho) [reply lib/AdobeState.js:341]: great maybe next time, as this is a custom error anyways

**Pull request merged**